Conceptpower+Spring/src/test/java/edu/asu/conceptpower/servlet/rest/ConceptSearchTest.java

```diff
@@ -0,0 +1,261 @@
+import edu.asu.conceptpower.app.util.URIHelper;
+import edu.asu.conceptpower.core.ConceptEntry;
+import edu.asu.conceptpower.core.ConceptType;
+import edu.asu.conceptpower.rest.ConceptSearch;
+import edu.asu.conceptpower.rest.ConceptSearchParameterValidator;
+import edu.asu.conceptpower.rest.ConceptSearchParameters;
+import edu.asu.conceptpower.rest.SearchParamters;
+import edu.asu.conceptpower.rest.msg.IMessageConverter;
+import edu.asu.conceptpower.rest.msg.IMessageRegistry;
+import edu.asu.conceptpower.rest.msg.json.JsonConceptMessage;
+import edu.asu.conceptpower.rest.msg.xml.XMLConceptMessage;
+import junit.framework.Assert;
+
+public class ConceptSearchTest {
+
+    @Mock
+    private IIndexService manager;
+
+    @Mock
+    private TypeDatabaseClient typeManager;
+
+    @Mock
+    private IMessageRegistry messageFactory;
+
+    @Mock
+    private URIHelper uriHelper;
+
+    @Mock
+    private BindingResult result;
+
+    @Mock
+    private ConceptSearchParameterValidator validator;
+
+    @InjectMocks
+    private ConceptSearch conceptSearch;
+
+    @Mock
+    private IMessageConverter xmlMessageFactory;
+
+    @Mock
+    private URIHelper uriCreator;
+
+    @Mock(name = "jsonMessageFactory")
+    private IMessageConverter jsonMessageFactory;
+
+    private ConceptSearchParameters conceptSearchParameter;
+
+    private String xmlNoRecordsFoundError = "<conceptpowerReply xmlns:digitalHPS=\"http://www.digitalhps.org/\"><digitalHPS:error error_message=\"No records found for the search condition.\" ></digitalHPS:error></conceptpowerReply>";
+    private int numberOfRecordsPerPage = 20;
+    private String jsonNoRecordsFoundError = "{\"message\":\"No records found for the search condition.\"}";
+
+    @Before
+    public void init() throws IllegalAccessException, LuceneException, IndexerRunningException {
+        MockitoAnnotations.initMocks(this);
+        Mockito.when(messageFactory.getMessageFactory(MediaType.APPLICATION_XML_VALUE)).thenReturn(xmlMessageFactory);
+        Mockito.when(messageFactory.getMessageFactory(MediaType.APPLICATION_XML_VALUE).createConceptMessage())
+                .thenReturn(new XMLConceptMessage(uriCreator));
+
+        Mockito.when(messageFactory.getMessageFactory(MediaType.APPLICATION_JSON_VALUE)).thenReturn(jsonMessageFactory);
+        Mockito.when(messageFactory.getMessageFactory(MediaType.APPLICATION_JSON_VALUE).createConceptMessage())
+                .thenReturn(new JsonConceptMessage(uriCreator));
+        conceptSearchParameter = new ConceptSearchParameters();
+        conceptSearchParameter.setPos("noun");
+        conceptSearchParameter.setWord("pony");
+        conceptSearchParameter.setConcept_list("Test-List");
+        conceptSearchParameter.setType_id("TYPE-123");
+        conceptSearchParameter.setOperator(SearchParamters.OP_AND);
+        conceptSearchParameter.setNumber_of_records_per_page(numberOfRecordsPerPage);
+
+        String typeId = "TYPE-123";
```

Comment:
all those strings should be in constants or final fields at the beginning of the method